### ISSUE TITLE
Circuit Design - Schematic Capture

### ISSUE THREAD

**Issue by zrajani** [zrajani/iLETTERBOX_CAPSTONE_PROJECT_EMBT_F2020#9]
After all the testing task has been accomplished one needs to sketch a plan on how the connections shall be made on the actual hardware. This is referred to as Schematic Capture.

**Comment by zrajani** [comment]
Creating a schematic capture for the connection we intend to make. Software confusion still exists between EASY EDA and FRITZING. Researching about which one is simpler to use. The schematics needs to be ready as soon as possible

**Comment by zrajani** [comment]
> Creating a schematic capture for the connection we intend to make. Software confusion still exists between EASY EDA and FRITZING. Researching about which one is simpler to use. The schematics need to be ready as soon as possible

Looking at the use and feasibility of the software use looks like there might be a hassle to use fritzing software on a Linux based operating system as they require the Qt5 Libraries to install. Also, I tried to install the software on the Ubuntu 18.05 LTS system and the Fritzing software looks to freeze-up and not function as per the expectation. Also, the newer version is priced and one cannot download without paying a minimum of 8 Euros (9 USD). Since we are not performing a large scale project and since other alternatives of open source are available. Finding libraries in Fritzing is difficult as compared to Easy EDA as the community support is good for Easy EDA as people contribute more in Easy EDA as compared to Fritzing 

On the other hand, the Easy EDA software was quite to easy to install on the Linux terminal and works very smoothly without any hassle. Also, many of the components are easily available and most of the features which are required by the project are at no cost. 

To conclude the point of selection of software to capture our schematic design I can say that since I have previously used Fritzing it's time to learn some new software which is more open-source and also will get me to know the alternative of the paid version software.

**Comment by zrajani** [comment]
EasyEDA software is quite good enough to have almost all of the components present but some of the components like the BB-WI were not present thus with help of the Beaglebone Black we edited the components as per our requirement. For example, we removed the Ethernet port from the Beaglebone Black as this is not present on the Beaglebone Black Wireless (BB-WI). Also, the Beaglebone Black is not available directly thus we imported the same using the Libary as we searched for Beaglebone and we got Beaglebone Black cape in the search result which we further edited as required for the project. The remaining components we are yet to find else we may have to edit some other components which may suit our project.

**Comment by zrajani** [comment]
Finding Components was troublesome but was resolved as had to check out the schematic for each one of them and then try to map them with the physical component present with us. But in the end, the circuit design is ready and shall be uploaded by today. The complete journey on how we found the libraries and the usage of the software can be found in the presentation pdf file present in the presentation folder of this repository

**Comment by zrajani** [comment]
I tried to generate the PCB design for the schematic we created but since the GSM Module had some issues as the pins on the schematic were not present in the schematic layout thus we switched to some other schematic which had the PCB pins arranged properly so PCB can be created for us and which can later be edited as per our requirement

**Comment by zrajani** [comment]
Currently, I have left the module which we choose earlier as the PCB was creating some issue since our main focus was to generate just the schematic capture I will leave that as research which will occur simultaneously and shall be completed when the deadline of the PCB Schematic arrives. Also yes one of more drawback of EasyEDA is is that it lacks the 3D and animated circuit parts which may be available in KiCad or Fritzing. But from a beginner point of view, this software is good enough though we understand that the animated or 3D view is more attractive to look and see as compared to the view which EasyEDA software provides. Maybe in the future, we shall try using and exploring some other software as we progress in the field and build some projects.

**Comment by mikeshams** [comment]
Zai - What issues did you face when working with EasyEDA?

**Comment by zrajani** [comment]
> Zain - What issues did you face when working with EasyEDA?

When working with the software I had some difficulties which were as follows [and these have been highlighted in the above messages as well]:

1. Finding Beaglebone WI was a problem as this was not directly present but since all the Beaglebone Boards are made based upon the Beaglebone Black. Thus, one can use the cape/ schematic of Beaglebone Black and edit as per the requirement for instance since the BB-WI has no ethernet port thus we just needed to delete the part from the schematic and then since the pins and other peripherals were the same thus we didn't have to delete or change them. 

2. The sensor pins sometimes didn't match thus we had to option browse some of the other available schematics or take the existing schematics and edit it. If we change the existing schematics we might have some issues that during the conversion of the PCB like the footprint or their respective setting might change. Thus for that reason, we opted to find the sensor schematic matching the physical component and then use it iff when it was not available then only we changed the schematic and contributed to the software back by saving the changes so others can use. 

3. Some of the schematics were small enough an it would be nice to have the option to resize them as this would get a uniformity in the design also look attractive. We tried using the attributes and other options to check if we can resize them but unfortunately we couldn't find any appropriate solution. 

4. When looking for the GSM Module which was looking identical to the way I have it physically I saw that the pins labelling was very dis-oriented thus had to orient it altogether which was quite irritating for me and time consuming. But cant blame the sofwtare for it as it was a User Contributed Library. 

Altogather it was a great journey exploring the software and yes would agree that tough it doesn't showcase much 3D schematic or the animated breadboard like in software like Fritizing but is best suited for anyone who wants to get started with schematic or PCB design. At the end of the task now I can work with two software namely Fritzing and EasyEDA but in the future I hope to explore and learn some of the other sofwtares as well and know how they tend to be different. For Feedback on the software I would it looks a kind of raw design if some animated designing or 3D viewing of the schematics can be added would be of great help so as to quickly find out if one must use it or no.
